schema/object.go

```diff
@@ -126,6 +132,22 @@ func (o *ObjectSchema) Unserialize(data any) (result any, err error) {
 	return rawData, nil
 }
 
+func (o *ObjectSchema) unserializeInlinedDataToMap(data any) (map[string]any, error) {
```

Comment:
This has the interesting property that it blindly *assumes* the map has a single key/value pair; if if given one with more it'll just return a map with the first key/value it happens to encounter. I suspect this behavior isn't useful, and I'd feel a bit better if it checked. (Granted, it's a private function and only currently called in one place where you've already checked; but still, explicit is better. If you don't want to check here, at least add a comment describing the intent of the deliberate design limitation for posterity.)

(And while not relevant to *this* PR, it also occurs to me that, like Ansible, you could allow inline JSON to say `map: {"field1": "value", "field2": "value"}` ... "eventually" if not now... an operation I think would be better suited to the implications of the function name! 😆 )

Reply:
I added the check (not pushed yet), but all it does is make sure you call the function in the correct scenario. So I made it panic instead of have a constraint error.